src/ipc-handlers.ts

```diff
@@ -674,39 +701,31 @@ export async function getThemeDetails(
 		throw new Error( 'Site not found.' );
 	}
 
-	if ( ! server.details.running || ! server.server ) {
-		return undefined;
-	}
+	const themeDetails = await server.getThemeDetails();
+	const themeChanged =
+		themeDetails?.path && themeDetails.path !== server.details.themeDetails?.path;
 
-	// CLI-managed sites don't have PHP instance, return cached theme details for Now
-	// ToDo: Implement logic to fetch theme details using mu-plugin?
-	if ( ! server.server.php ) {
-		return server.details.themeDetails;
-	}
-
-	const themeDetails = await phpGetThemeDetails( server.server );
-
-	const parentWindow = BrowserWindow.fromWebContents( event.sender );
-	if ( themeDetails?.path && themeDetails.path !== server.details.themeDetails?.path ) {
+	if ( themeChanged ) {
+		const parentWindow = BrowserWindow.fromWebContents( event.sender );
 		sendIpcEventToRendererWithWindow( parentWindow, 'theme-details-updating', { id } );
-		const updatedSite = {
-			...server.details,
-			themeDetails,
-		};
 		sendIpcEventToRendererWithWindow( parentWindow, 'theme-details-changed', {
 			id,
 			details: themeDetails,
 		} );
 
-		void server
-			.updateCachedThumbnail()
-			.then( () => sendThumbnailChangedEvent( event, id ) )
-			.catch( ( error ) => {
-				console.error( `Failed to update thumbnail for server ${ id }:`, error );
-			} );
 		server.details.themeDetails = themeDetails;
-		await updateSite( event, updatedSite );
+		await updateSite( event, server.details );
+
+		void ( async () => {
+			try {
+				await server.updateCachedThumbnail();
+				await sendThumbnailChangedEvent( event, id );
+			} catch ( error ) {
+				console.error( `Failed to update thumbnail for server ${ id }:`, error );
+			}
+		} )();
```

Comment:
```suggestion
		try {
			await server.updateCachedThumbnail();
			await sendThumbnailChangedEvent( event, id );
		} catch ( error ) {
			console.error( `Failed to update thumbnail for server ${ id }:`, error );
		}
```
This has the same effect. We just don't want to propagate any exceptions

Reply:
We also don't want to await the thumbnail update, but need those 2 operations to happen in sequence. I rewrote it using .then() since I believe it makes the intention easier to read.